### DISCUSSION TITLE
Lesson6: Segmentation fault error while deploying simpleStorage contract using anvil

### DISCUSSION THREAD

**Discussion by anb5** [Cyfrin/foundry-full-course-cu#844] [Q&A]
I'm trying to deploy the simplestorage contract using Anvil but it's not getting deployed. Below are the steps I followed

1. I started anvil and added a local network in the metamask
2. I opened another terminal and ran this command "forge create SimpleStorage --rpc-url http://127.0.0.1:8545 --interactive"
3. It is asking for the private key I entered it
After this, the anvil closes itself and the forge terminal freezes. 

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/117075012/7512212f-93dd-4c83-8632-682f7e39ffe8)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/117075012/c946525e-09f5-404e-8923-d92fe1a55301)

**Comment by theirrationalone** [comment]
Heyy @PatrickAlphaC !, Is it a BUG 👾 ???

**Comment by Jamster187** [comment]
I'm not 100% if this is your issue, but the terminal will not seem "responsive" when it is asking for your private key. Just copy + paste + enter and it should work. Hopefully that solves your problem.

**Reply by anb5** [reply]
I'm able to enter the private and moreover I've tried to give private key in the forge command like --private-key 0xxxxx but still it is giving segmentation fault error

**Reply by PatrickAlphaC** [reply]
Could you log this on the foundry github? 

https://github.com/foundry-rs/foundry

**Reply by anb5** [reply]
Sure will do. Meanwhile, can you tell me any alternatives for anvil

**Comment by theirrationalone** [comment]
> I'm not 100% if this is your issue, but the terminal will not seem "responsive" when it is asking for your private key. Just copy + paste + enter and it should work. Hopefully that solves your problem.

It's an unexpected error/fault/bug.
